app/services/api_authentication_service.rb

```diff
@@ -0,0 +1,16 @@
+class APIAuthenticationService
+  def self.identify_user(request)
+    if token = given_token(request)
+      matched_token = APIToken.active.where(token: token).first
+      matched_token&.user
+    end
+  end
+
+  def self.authorization_given?(request)
+    request.headers['Authorization'].present?
+  end
+
+  def self.given_token(request)
+    request.headers['Authorization'].to_s.gsub(/Bearer\s+([^\s]*)$/, '\1')
+  end
```

Comment:
It feels like this class is mixing two concerns:

1. Extracting the bearer token from the request.
2. Identifying the user whose token it is

IMHO, the extraction of the bearer token should be a method on `Computacenter::API::BaseController` because it's related to translating from HTTP requests to higher-level data, which is a controller responsibility:

```
def bearer_token
  request.headers['Authorization']&.to_s&.gsub(/Bearer\s+([^\s]*)$/, '\1')
end
```

Then this service simply becomes:

```
class APIAuthenticationService
  def self.identify_user(token)
    if matched_token = APIToken.active.where(token: token).first
      matched_token&.user
    end
  end
end
```
and you call `APIAuthenticationService.identify_user(bearer_token)` in the controller.

WDYT?

Reply:
Actually it could even be

```
class APIAuthenticationService
  def self.identify_user(token)
    APIToken.active.find_by(token: token)&.user
  end
end
```

Reply:
It's a fair point, and a good idea... except that we need to call this from `ApplicationController`. So we'd then need to override the `populate_user_from_session_or_api_token!` method in the Computacenter::BaseController, which also feels a bit messy in a different way.

I'll try a few more things

Reply:
> It's a fair point, and a good idea... except that we need to call this from ApplicationController

`bearer_token` could sit in `ApplicationController` so that problem goes away...

Reply:
Yes! If I also -
* rename the `populate_user_from_session_or_api_token!` method to `identify_user!`
* put it back to just considering the user session
* override it in the computacenter/api/base_controller
- then that works, and looks a lot cleaner too. Thanks!

_(Naming Things Right Is Hard But Often Crucial part 174825)_